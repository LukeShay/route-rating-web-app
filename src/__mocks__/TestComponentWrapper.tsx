/* eslint @typescript-eslint/no-empty-function: 0, @typescript-eslint/explicit-function-return-type: 0 */
import React from "react";
import { IGymsContextState } from "../context/gyms/gymsStore";
import { IViewContextState } from "../context/view/viewStore";
import { IUserContextState } from "../context/user/userStore";
import { StaticRouter } from "react-router";
import GymsStoreMock from "./gymsStoreMock";
import ViewStoreMock from "./viewStoreMock";
import UserStoreMock from "./userStoreMock";

export interface ITestComponentWrapperProps {
  children: React.ReactNode;
  gymsState: IGymsContextState;
  location: string;
  userState: IUserContextState;
  viewState: IViewContextState;
}

const TestComponentWrapper: React.FC<ITestComponentWrapperProps> = ({
  children,
  gymsState,
  location,
  userState,
  viewState
}): JSX.Element => {
  return (
    <StaticRouter location={`https://lukeshay.com/${location}`}>
      <GymsStoreMock state={gymsState} dispatch={() => {}}>
        <ViewStoreMock state={viewState} dispatch={() => {}}>
          <UserStoreMock state={userState} dispatch={() => {}}>
            {children}
          </UserStoreMock>
        </ViewStoreMock>
      </GymsStoreMock>
    </StaticRouter>
  );
};

export default TestComponentWrapper;
