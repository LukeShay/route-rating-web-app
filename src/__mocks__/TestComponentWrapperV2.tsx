/* eslint @typescript-eslint/no-empty-function: 0, @typescript-eslint/explicit-function-return-type: 0 */
import React from "react";
import { IGymsContextState } from "../context/gyms/gymsStore";
import { IUserContextState } from "../context/user/userStore";
import { IViewContextState } from "../context/view/viewStore";
import GymsStoreMock from "./gymsStoreMock";
import ViewStoreMock from "./viewStoreMock";
import UserStoreMock from "./userStoreMock";

export interface ITestComponentWrapperV2Props {
  children: React.ReactNode;
  gymsState: IGymsContextState;
  userState: IUserContextState;
  viewState: IViewContextState;
}

const TestComponentWrapperV2: React.FC<ITestComponentWrapperV2Props> = ({
  children,
  gymsState,
  userState,
  viewState
}): JSX.Element => {
  return (
    <GymsStoreMock state={gymsState} dispatch={() => {}}>
      <ViewStoreMock state={viewState} dispatch={() => {}}>
        <UserStoreMock state={userState} dispatch={() => {}}>
          {children}
        </UserStoreMock>
      </ViewStoreMock>
    </GymsStoreMock>
  );
};

export default TestComponentWrapperV2;
