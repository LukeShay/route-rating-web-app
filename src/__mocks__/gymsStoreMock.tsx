import { GymsContext, IGymsContextState } from '../context/gyms/gymsStore';
import React from 'react';

export interface IGymsStoreMockProps {
  state: IGymsContextState;
  dispatch: any;
}

const GymsStoreMock: React.FC<IGymsStoreMockProps> = ({
  state,
  dispatch,
  children,
}): JSX.Element => (
  <GymsContext.Provider value={{ state, dispatch }}>
    {children}
  </GymsContext.Provider>
);

export default GymsStoreMock;
