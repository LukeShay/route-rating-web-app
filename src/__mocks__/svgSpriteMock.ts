export default {
  id: 'svg-sprite-id',
  viewBox: '0 0 1 1',
  url: '/sprite.svg#svg-sprite-id',
};
