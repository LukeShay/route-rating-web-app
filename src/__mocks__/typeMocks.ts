import { Gym, GymPageType, Route, User, Wall, WallTypes } from '../types';
import { GymsContextState } from '../context/gyms/gymsStore';
import { ViewContextState } from '../context/view/viewStore';
import { UserContextState } from '../context/user/userStore';

export const testRouteOne: Route = {
  averageGrade: 'GRADE_5_9',
  averageRating: 4,
  gymId: 'gym1',
  holdColor: 'Blue',
  id: 'route1',
  name: 'RouteOne',
  setter: 'LukeOne',
  types: [WallTypes.TOP_ROPE, WallTypes.LEAD],
  wallId: 'wall1',
};

export const testRouteTwo: Route = {
  averageGrade: 'GRADE_5_12ab',
  averageRating: 4,
  gymId: 'gym1',
  holdColor: 'Red',
  id: 'route2',
  name: 'RouteTwo',
  setter: 'LukeTwo',
  types: [WallTypes.TOP_ROPE, WallTypes.LEAD],
  wallId: 'wall1',
};

export const testWallOne: Wall = {
  gymId: 'gym1',
  id: 'wall1',
  name: 'Wall1',
  routes: [testRouteOne, testRouteTwo],
  types: [WallTypes.LEAD, WallTypes.TOP_ROPE],
};

export const testWallTwo: Wall = {
  gymId: 'gym1',
  id: 'wall2',
  name: 'Wall2',
  routes: [],
  types: [WallTypes.LEAD, WallTypes.TOP_ROPE],
};

export const testGymOne: Gym = {
  address: '1234 Five Street',
  authorizedEditors: ['id'],
  city: 'Ames',
  email: 'abc@d.com',
  id: 'gym1',
  logoUrl: '',
  name: 'Gym One',
  phoneNumber: '1234567890',
  photoUrl: '',
  state: 'Iowa',
  walls: [testWallOne, testWallTwo],
  website: 'lukeshay.com',
  zipCode: '50014',
};

export const testGymTwo: Gym = {
  address: '1234 Five Street',
  authorizedEditors: ['id'],
  city: 'Ames',
  email: 'abc@d.com',
  id: 'gym2',
  logoUrl: '',
  name: 'Gym Two',
  phoneNumber: '1234567890',
  photoUrl: '',
  state: 'Iowa',
  walls: [testWallOne, testWallTwo],
  website: 'lukeshay.com',
  zipCode: '50014',
};

export const testUserEditor: User = {
  authority: 'ADMIN',
  city: 'Ames',
  country: 'USA',
  email: 'email',
  firstName: 'Name',
  id: 'id',
  lastName: 'Last',
  password: 'password',
  phoneNumber: '1111111111',
  role: 'ADMIN_ROLE',
  session: null,
  state: 'IA',
  username: 'username',
};

export const mockGymsState: GymsContextState = {
  page: { content: [testGymOne, testGymTwo] } as GymPageType,
};

export const darkDesktopState: ViewContextState = {
  mobile: false,
  theme: 'DARK_THEME',
};

export const signedInEditorState: UserContextState = {
  user: testUserEditor,
};

export const signedOutState: UserContextState = {
  user: null,
};
