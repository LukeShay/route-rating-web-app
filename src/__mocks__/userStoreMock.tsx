import { IUserContextState, UserContext } from '../context/user/userStore';
import React from 'react';

export interface IUserStoreMockProps {
  state: IUserContextState;
  dispatch: any;
}

const UserStoreMock: React.FC<IUserStoreMockProps> = ({
  state,
  dispatch,
  children,
}): JSX.Element => (
  <UserContext.Provider value={{ state, dispatch }}>
    {children}
  </UserContext.Provider>
);

export default UserStoreMock;
