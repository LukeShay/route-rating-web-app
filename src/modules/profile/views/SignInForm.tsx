import LockOutlinedIcon from '@material-ui/icons/LockOutlined';
import React from 'react';
import * as UserActions from '../../../context/user/userActions';
import { useUserContext } from '../../../context/user/userStore';
import Button from '../../common/buttons/ButtonSecondary';
import Form from '../../common/forms/Form';
import CheckBox from '../../common/inputs/CheckBox';
import Input from '../../common/inputs/Input';
import { ButtonEvent, InputEvent } from '../../../types';

export interface PropsLogInForm {
  handleSignUpClick(event: ButtonEvent): void;
}

const SignInForm: React.FC<PropsLogInForm> = (props): JSX.Element => {
  const { dispatch: userDispatch } = useUserContext();
  const [username, setUsername] = React.useState<string>('');
  const [password, setPassword] = React.useState<string>('');
  const [passwordMessage, setPasswordMessage] = React.useState<string>('');
  const [rememberMe, setRememberMe] = React.useState<boolean>(false);

  const handleChange = async (event: InputEvent): Promise<void> => {
    const { id, value } = event.target;

    if (id === 'username') {
      setUsername(value);
    } else if (id === 'password') {
      setPassword(value);
    } else if (id === 'rememberMe') {
      setRememberMe(!rememberMe);
    }
  };

  const handleSubmit = (event: ButtonEvent): void => {
    event.preventDefault();

    UserActions.signIn(userDispatch, username, password, rememberMe).then(
      (response) => {
        if (response instanceof Response && response.status === 401) {
          setPasswordMessage(
            'User not found or incorrect password. Try a different username or password.'
          );
        } else if (!(response instanceof Response)) {
          setPasswordMessage('There was an error. Please try again.');
        }
      }
    );
  };

  const formInputs: JSX.Element = (
    <React.Fragment>
      <Input
        placeholder="Username"
        id="username"
        value={username}
        onChange={handleChange}
        type="text"
        autoComplete="email"
      />
      <Input
        placeholder="Password"
        id="password"
        value={password}
        onChange={handleChange}
        helpText={passwordMessage}
        type="password"
        autoComplete="current-password"
      />
      <CheckBox
        checked={rememberMe}
        value="REMEMBER_ME"
        onChange={handleChange}
        id="rememberMe"
        label="Remember Me"
      />
    </React.Fragment>
  );

  const title: JSX.Element = (
    <div style={{ display: 'inline' }}>
      <div style={{ float: 'left', marginRight: '25px', marginTop: '5px' }}>
        Sign in
      </div>
      <div style={{ float: 'right', marginLeft: '25px' }}>
        <Button
          id="signUp"
          onClick={props.handleSignUpClick}
          type="button"
          variant="outlined"
        >
          Sign up
        </Button>
      </div>
    </div>
  );

  return (
    <Form
      id="signIn"
      buttonText="Sign in"
      formInputs={formInputs}
      handleSubmit={handleSubmit}
      icon={<LockOutlinedIcon />}
      title={title}
    />
  );
};

export default SignInForm;
