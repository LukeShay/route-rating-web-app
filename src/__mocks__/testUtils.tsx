import { render, RenderResult } from '@testing-library/react';
import React from 'react';
import TestComponentWrapperV2 from './TestComponentWrapperV2';
import * as TypeMocks from './typeMocks';

interface WrapperProps {
  children: React.ReactNode;
}

const SignedInDesktop: React.FC<WrapperProps> = ({ children }): JSX.Element => {
  return (
    <TestComponentWrapperV2
      gymsState={TypeMocks.mockGymsState}
      userState={TypeMocks.signedInEditorState}
      viewState={TypeMocks.darkDesktopState}
    >
      {children}
    </TestComponentWrapperV2>
  );
};

export const renderSignedInDesktop = (ui, options): RenderResult =>
  render(ui, { wrapper: SignedInDesktop, ...options });

const SignedOutDesktop: React.FC<WrapperProps> = ({
  children,
}): JSX.Element => {
  return (
    <TestComponentWrapperV2
      gymsState={TypeMocks.mockGymsState}
      userState={TypeMocks.signedOutState}
      viewState={TypeMocks.darkDesktopState}
    >
      {children}
    </TestComponentWrapperV2>
  );
};

export const renderSignedOutDesktop = (ui, options): RenderResult =>
  render(ui, { wrapper: SignedOutDesktop, ...options });

export * from '@testing-library/react';
