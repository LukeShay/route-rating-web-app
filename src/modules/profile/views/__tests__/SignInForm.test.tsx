import React from 'react';
import SignInForm from '../SignInForm';
import { shallow } from '../../../../__mocks__/setupEnzyme';
import { shallowToJson } from 'enzyme-to-json';

describe('SignInForm', () => {
  it('should render correctly.', async () => {
    const wrapper = shallow(<SignInForm handleSignUpClick={jest.fn()} />);
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
