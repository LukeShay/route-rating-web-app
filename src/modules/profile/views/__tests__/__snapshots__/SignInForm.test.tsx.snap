// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignInForm should render correctly. 1`] = `
<Form
  buttonText="Sign in"
  formInputs={
    <React.Fragment>
      <Input
        autoComplete="email"
        error={true}
        fullWidth={true}
        helpText=""
        id="username"
        name=""
        onChange={[Function]}
        placeholder="Username"
        type="text"
        value=""
      />
      <Input
        autoComplete="current-password"
        error={true}
        fullWidth={true}
        helpText=""
        id="password"
        name=""
        onChange={[Function]}
        placeholder="Password"
        type="password"
        value=""
      />
      <CheckBox
        checked={false}
        color="default"
        id="rememberMe"
        label="Remember Me"
        onChange={[Function]}
        value="REMEMBER_ME"
      />
    </React.Fragment>
  }
  handleSubmit={[Function]}
  icon={<Memo(LockOutlinedIcon) />}
  id="signIn"
  title={
    <div
      style={
        Object {
          "display": "inline",
        }
      }
    >
      <div
        style={
          Object {
            "float": "left",
            "marginRight": "25px",
            "marginTop": "5px",
          }
        }
      >
        Sign in
      </div>
      <div
        style={
          Object {
            "float": "right",
            "marginLeft": "25px",
          }
        }
      >
        <ButtonSecondary
          disabled={false}
          fullWidth={true}
          id="signUp"
          onClick={[MockFunction]}
          size="medium"
          type="button"
          variant="outlined"
        >
          Sign up
        </ButtonSecondary>
      </div>
    </div>
  }
/>
`;
